src/features/auth/pages/forgottenPassword/ReinitializePassword/ReinitializePassword.native.test.tsx

```diff
@@ -257,12 +275,23 @@ describe('ReinitializePassword Page', () => {
     await act(async () => {})
 
     expect(replace).toHaveBeenNthCalledWith(1, 'ResetPasswordExpiredLink', {
-      email: ROUTE_PARAMS.email,
+      email: emailFromParams,
     })
   })
 })
 
-function renderReinitializePassword() {
+const emailFromParams = 'john@.example.com'
+const tokenFromParams = 'reerereskjlmkdlsf'
+
+function renderReinitializePassword(from?: Referrals) {
+  useRoute.mockImplementation(() => ({
+    params: {
+      email: emailFromParams,
+      token: tokenFromParams,
+      expiration_timestamp: 45465546445,
+      from,
+    },
+  }))
```

Comment:
Je préfère mettre le mock en dehors du render, ça permettait de mocker qu'une seule fois.
Si tu le mets dans le render parce que le mock "ne fonctionnait pas" avec un mockReturnValueOnce, tu peux utiliser le `mockReturnValueOnce` et expliqué que c'est à cause des nombreux renders

Reply:
Si j'ai fait ça, c'est parce que je n'arrive pas a modifier la valeur de 'from' autrement, pour tester un cas 'undefined' et un cas ou il a une valeur, je sais pas trop pourquoi :(

Reply:
Le composant est render 14 fois pendant le test du coup il faudrait le mettre vraiment plein de fois pour que ça passe ^^ Je te propose de laisser comme t'as fait maintenant, en "remettant" la valeur par défaut dans un `beforeEach`

Reply:
Mais ce que j'ai fait n'est pas bon, parce que je ne peux pas utiliser mockReturnValue dans un it et mockReturnValueOnce ne fonctionne pas :(

Reply:
Tu peux utiliser `// eslint-disable-next-line local-rules/independent-mocks` pour ignorer l'erreur de linter (c'est déjà utilisé pleins de fois dans ce fichier). Sinon, tu peux modifier l'object : 
```ROUTE_PARAMS.from = "blabla" ```
et remettre à undefined dans le `beforeEach` (il faudra typer `ROUTE_PARAMS` pour que eslint ne râle pas.